### PR TITLE
Issue #7579: Update doc for AvoidNestedBlocks

### PR DESCRIPTION
**Pull request by bossever**
Issue #7579: Added code examples for AvoidNestedBlocks
Moved the example for allowInSwitchCase property from the Description sub-section to Examples. 
image

Output of default config examples:

`$ cat custom_config.xml`
```
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
  <module name="Checker">
    <module name="TreeWalker">
    <module name="AvoidNestedBlocks"/>
  </module>
</module>
```

`$ cat Test.java`
```
public void foo() {
  int myInteger = 0;
  {
    myInteger = 2;    //NOT OK, violation : Avoid nested blocks.
  }
  System.out.println("myInteger = " + myInteger);

  switch (a) {
    case 1:
      {
        System.out.println("Case 1");   //NOT OK, violation : Avoid nested blocks.
        break;
      }
    case 2:
      System.out.println("Case 2");     //OK
      break;
  }
}
```

`$ java com.puppycrawl.tools.checkstyle.Main -c custom_config.xml Test.java`
```
Starting audit...
[ERROR] /home/ayushman/Desktop/CheckstyleExperiments/Test.java:3:3: Avoid nested blocks. [AvoidNestedBlocks]
[ERROR] /home/ayushman/Desktop/CheckstyleExperiments/Test.java:10:7: Avoid nested blocks. [AvoidNestedBlocks]
Audit done.
Checkstyle ends with 2 errors.
```

Output of non-default config examples:

`$ cat custom_config.xml`
```
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
  <module name="Checker">
    <module name="TreeWalker">
    <module name="AvoidNestedBlocks">
      <property name="allowInSwitchCase" value="true"/>
    </module>
  </module>
</module>
```

`$ cat Test.java`
```
switch (a)
{
  case 0:
    // Never OK, break outside block
    {
      x = 1;
    }
    break;
  case 1:
    // Never OK, statement outside block
    System.out.println("Hello");
    {
      x = 2;
      break;
    }
  case 2:
    // OK if allowInSwitchCase is true
    {
      System.out.println("Hello");
      x = 3;
      break;
    }
}
```

`$ java com.puppycrawl.tools.checkstyle.Main -c custom_config.xml Test.java`
```
Starting audit...
[ERROR] /home/ayushman/Desktop/CheckstyleExperiments/Test.java:5:5: Avoid nested blocks. [AvoidNestedBlocks]
[ERROR] /home/ayushman/Desktop/CheckstyleExperiments/Test.java:12:5: Avoid nested blocks. [AvoidNestedBlocks]
Audit done.
Checkstyle ends with 2 errors.
```

### CODE REVIEW

**Comment by strkkk** [comment]
@bossever I have restarted failed tasks

**Comment by bossever** [comment]
@strkkk Please review the PR, all checks have passed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 1]
```diff
@@ -61,6 +61,58 @@
...
+ * public void foo() {
+ *   int myInteger = 0;
+ *   {
+ *     myInteger = 2;    //NOT OK, violation : Avoid nested blocks.
+ *   }
+ *   System.out.println("myInteger = " + myInteger);
+ * }
+ * </pre>
+ * <p>
+ * Example of redundant nested block in switch case:
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:91]: Both examples(this and above) are for same default config, please merge them into one

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 2]
```diff
@@ -61,6 +61,58 @@
...
+ * &lt;module name="AvoidNestedBlocks"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public void foo() {
+ *   int myInteger = 0;
+ *   {
+ *     myInteger = 2;    //NOT OK, violation : Avoid nested blocks.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:85]: `//NOT OK, violation : Avoid nested blocks.` just `// violation` is enough.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 3]
```diff
@@ -61,6 +61,58 @@
...
+ * }
+ * </pre>
+ * <p>
+ * Example of redundant nested block in switch case:
+ * </p>
+ * <pre>
+ * switch (a) {
+ *   case 1:
+ *     {
+ *       System.out.println("Case 1");   //NOT OK, violation : Avoid nested blocks.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:97]: same here

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 4]
```diff
@@ -61,6 +61,58 @@
...
+ *     System.out.println("Case 2");     //OK
+ *     break;
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to allow nested blocks in switch case:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AvoidNestedBlocks&quot;&gt;
+ * &lt;property name=&quot;allowInSwitchCase&quot; value=&quot;true&quot;/&gt;
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:110]: indentation is missing

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 2 above)
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:85]: violation comment should be on line where violation is reported, which is line above

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 5]
```diff
@@ -61,6 +61,58 @@
...
+ * </p>
+ * <pre>
+ * switch (a) {
+ *   case 1:
+ *     {
+ *       System.out.println("Case 1");   //NOT OK, violation : Avoid nested blocks.
+ *       break;
+ *     }
+ *   case 2:
+ *     System.out.println("Case 2");     //OK
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:101]: `//OK` --> `// OK`

**Review by strkkk — CHANGES_REQUESTED** [review]

**Comment by bossever** [comment]
@strkkk I have made the requested changes.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 6]
```diff
@@ -61,6 +61,55 @@
...
+ * <pre>
+ * public void foo() {
+ *   int myInteger = 0;
+ *   {                      // violation
+ *     myInteger = 2;
+ *   }
+ *   System.out.println("myInteger = " + myInteger);
+ * }
+ *
+ * //example of redundant nested block in switch case
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:90]: no need for this comment. Please move this switch example to method above to make it compliable

File: src/xdocs/config_blocks.xml [hunk 7]
```diff
@@ -110,6 +85,63 @@ switch (a)
         <source>
 &lt;module name=&quot;AvoidNestedBlocks&quot;/&gt;
         </source>
+
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:88]: extra line not needed

File: src/xdocs/config_blocks.xml [hunk 8]
```diff
@@ -110,6 +85,63 @@ switch (a)
...
+}
+        </source>
+
+        <p> To configure the check to allow nested blocks in switch case: </p>
+        <source>
+&lt;module name=&quot;AvoidNestedBlocks&quot;&gt;
+  &lt;property name=&quot;allowInSwitchCase&quot; value=&quot;true&quot;/&gt;
+&lt;/module&gt;
+        </source>
+
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:118]: same here

File: src/xdocs/config_blocks.xml [hunk 9]
```diff
@@ -110,6 +85,63 @@ switch (a)
         <source>
 &lt;module name=&quot;AvoidNestedBlocks&quot;/&gt;
         </source>
+
+        <p> Example: </p>
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:89]: no need for extra leading/trailing spaces

File: src/xdocs/config_blocks.xml [hunk 10]
```diff
@@ -110,6 +85,63 @@ switch (a)
...
+      System.out.println("Case 1");
+      break;
+    }
+  case 2:
+    System.out.println("Case 2");     // OK
+    break;
+}
+        </source>
+
+        <p> To configure the check to allow nested blocks in switch case: </p>
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:112]: same here

File: src/xdocs/config_blocks.xml [hunk 11]
```diff
@@ -110,6 +85,63 @@ switch (a)
...
+    {                   // violation
+      System.out.println("Case 1");
+      break;
+    }
+  case 2:
+    System.out.println("Case 2");     // OK
+    break;
+}
+        </source>
+
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:111]: same here

File: src/xdocs/config_blocks.xml [hunk 12]
```diff
@@ -110,6 +85,63 @@ switch (a)
...
+        <p> To configure the check to allow nested blocks in switch case: </p>
+        <source>
+&lt;module name=&quot;AvoidNestedBlocks&quot;&gt;
+  &lt;property name=&quot;allowInSwitchCase&quot; value=&quot;true&quot;/&gt;
+&lt;/module&gt;
+        </source>
+
+        <p> Example: </p>
```
Comment (strkkk) [inline src/xdocs/config_blocks.xml:119]: same here

**Review by strkkk — CHANGES_REQUESTED** [review]

**Comment by strkkk** [comment]
@bossever Please 
1. Do not close/reopen PRs
2. Do not resolve review items, it is up to reviewer to decide is ok or not. Just reply "done" when you made required changes

**Comment by bossever** [comment]
> @bossever Please 
> 1. Do not close/reopen PRs
> 2. Do not resolve review items, it is up to reviewer to decide is ok or not. Just reply "done" when you made required changes 

1. Yes, I didn't mean to do that
2. Okay, will keep in mind.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 6 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/xdocs/config_blocks.xml (hunk 7 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

File: src/xdocs/config_blocks.xml (hunk 8 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

File: src/xdocs/config_blocks.xml (hunk 9 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

File: src/xdocs/config_blocks.xml (hunk 10 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

File: src/xdocs/config_blocks.xml (hunk 11 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

File: src/xdocs/config_blocks.xml (hunk 12 above)
Comment (bossever) [reply src/xdocs/config_blocks.xml]: Done

**Comment by romani** [comment]
@bossever , please keep in this PR ONLY single commit `Issue #7579: Update doc for AvoidNestedBlocks` all other should not be there.
if you do not know how to fix it.
1) get latest upstream master branhc
2) create new branch
3) manaully move all changes to new branch (single commit)
4) remove old/exiting branch
5) rename new branch to old branch.
6) push to remote with force.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 1 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 2 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 3 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 4 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 2 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 5 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Done

**Review by strkkk — APPROVED** [review]

**Review by pbludov — APPROVED** [review]
I'm ok to merge.

**Comment by romani** [comment]
Github, generate web site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 13]
```diff
@@ -61,6 +61,54 @@
...
+ * <pre>
+ * &lt;module name="AvoidNestedBlocks"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public void foo() {
+ *   int myInteger = 0;
+ *   {                      // violation
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:84]: good example , and please use whole example as-is in seconds code block.
https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples see block "ATTENTION" for a reason.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 13 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:84]: Done, now both examples are almost same.

**Comment by romani** [comment]
Github, generate web site

**Comment by romani** [comment]
Github, generate web site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java [hunk 14]
```diff
@@ -101,6 +76,67 @@
...
+ *     case 1:
+ *       {                    // OK
+ *         System.out.println("Case 1");
+ *         break;
+ *       }
+ *     case 2:
+ *       System.out.println("Case 2");     // OK
+ *       break;
+ *     case 3:
+ *       // Never OK, statement outside block
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java:131]: please remove "case 3" this comment specifically blow my mind and will definitely do confusion for users.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java (hunk 14 above)
Comment (bossever) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java]: Ok, done.

**Review by romani — APPROVED** [review]
ok to merge if CI green.

@bossever , thanks a lot for contribution.

**Comment by romani** [comment]
Github, generate web site

**Comment by romani** [comment]
report looks good

**Pull request merged**